backend/de.metas.contracts/src/main/java/de/metas/contracts/impl/FlatrateBL.java

```diff
@@ -2674,4 +2683,27 @@ public void reverseDefinitiveInvoice(@NonNull final Collection<FlatrateTermId> c
 	{
 		flatrateDAO.reverseDefinitiveInvoice(contractIds);
 	}
+
+	@Override
+	@NonNull
+	public CurrencyPrecision getPricePrecisionForModularContract(@NonNull final FlatrateTermId modularContractId)
+	{
+		final I_C_Flatrate_Term term = getById(modularContractId);
+		final ConditionsId conditionsId = ConditionsId.ofRepoId(term.getC_Flatrate_Conditions_ID());
+
+		Check.assume(isModularContract(conditionsId) || isInterimContract(conditionsId), "The contract must be modular on interim.");
+
+		final ZonedDateTime date = TimeUtil.asZonedDateTime(term.getStartDate(), orgDAO.getTimeZone(OrgId.ofRepoId(term.getAD_Org_ID())));
+		final ModularContractSettings settings = modularContractSettingsRepository.getByFlatrateTermId(modularContractId);
+		final I_M_PriceList priceList = priceListBL.getCurrentPricelistOrNull(
+				settings.getPricingSystemId(),
+				bPartnerDAO.getCountryId(BPartnerLocationId.ofRepoId(term.getBill_BPartner_ID(), term.getBill_Location_ID())),
+				date,
+				settings.getSoTrx()
+		);
+
+		final CurrencyPrecision currencyPrecision = currencyBL.getStdPrecision(CurrencyId.ofRepoId(term.getC_Currency_ID()));
```

Comment:
just a minor idea..what about avoiding to get the currency's precision when we found that it can be taken from the pricelist?

Reply:
Good point! Fixed it.